new-log-viewer/src/services/LogFileManager.ts

```diff
@@ -195,14 +196,15 @@ class LogFileManager {
             };
         }
 
-        let beginLogEventIdx: number = 1;
+        let beginLogEventIdx: number = 0;
         const {code, args} = cursor;
         if (CURSOR_CODE.PAGE_NUM === code) {
             beginLogEventIdx = ((args.pageNum - 1) * this.#pageSize);
         }
         if (CURSOR_CODE.LAST_EVENT === code || beginLogEventIdx > this.#numEvents) {
+            // Set to the first event of the last page
             beginLogEventIdx =
-                    (Math.floor((this.#numEvents - 1) / this.#pageSize) * this.#pageSize);
+                getPageNumFromLogEventNum(this.#numEvents, this.#pageSize) * this.#pageSize;
```

Comment:
Sorry for missing this in the previous change.

Do you think it would be more straight forward to write
```sugguestion
(getPageNumFromLogEventNum(this.#numEvents, this.#pageSize) - 1) * this.#pageSize;
```
?

Essentially, `getPageNumFromLogEventNum()` returns a "num" (1-based index). To compute `beginLogEventIdx` which is a 0-based index, we want to convert the 1-based index to a 0-based index before we multiply it with the page size, so we subtract `getPageNumFromLogEventNum(this.#numEvents, this.#pageSize)` by 1 before that.

Reply:
Sure